File: 프로그래머스/lv2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    unordered_map<string, int> m;
    for (auto cloth : clothes)
        m[cloth[1]]++;
    for (auto elem : m)
        answer *= elem.second + 1;
    return answer - 1;
}

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int solution(String[][] clothes) {
        Map<String, Integer> map = new HashMap<>();
        for (String[] cloth : clothes)
            map.put(cloth[1], map.getOrDefault(cloth[1], 0) + 1);
        return map.values().stream()
            .map(a -> a + 1).reduce((a, b) -> a * b).orElse(0) - 1;
    }
}